sdk/src/main/java/com/sygic/travel/sdk/auth/facade/AuthFacade.kt

```diff
@@ -0,0 +1,170 @@
+package com.sygic.travel.sdk.auth.facade
+
+import com.sygic.travel.sdk.Callback
+import com.sygic.travel.sdk.auth.model.AuthorizationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationRequest
+import com.sygic.travel.sdk.auth.model.UserRegistrationResponse
+import com.sygic.travel.sdk.auth.model.UserSession
+import com.sygic.travel.sdk.auth.service.AuthService
+import com.sygic.travel.sdk.auth.service.AuthStorageService
+import com.sygic.travel.sdk.utils.runAsync
+import com.sygic.travel.sdk.utils.runWithCallback
+
+
+class AuthFacade(
+	private val authService: AuthService,
+	private val authStorageService: AuthStorageService
+) {
+
+	companion object {
+		const val CLIENT_ID = "sygictravel_android_sdk_demo"
+		const val DEVICE_PLATFORM = "android"
+	}
+
+	fun loginUserWithPassword(
+		username: String,
+		password: String,
+		callback: Callback<UserSession?>
+	) {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		runWithCallback({ authService.authorize(request) }, callback)
+	}
+
+	suspend fun loginUserWithPassword(username: String, password: String): UserSession? {
+		val request = RequestBuilder.passwordAuthRequest(username, password)
+		return runAsync { authService.authorize(request) }
+	}
+
+	fun loginUserByGoogle(googleToken: String?, callback: Callback<UserSession?>) {
```

Comment:
Same problem with name `loginUserByGoogle` => `loginUserWithGoogle`

Reply:
@costasovo correctly, you should update the interface in specification, otherwise you will have to review this 3 times :D

Reply:
Sure I updated it here https://confluence.sygic.com/display/STV/B2B+-+User+sessions right after first review
We should add the section also to SDK modules. Any volunteer?